backend/papersfeed/utils/recommendations/utils.py

```diff
@@ -0,0 +1,60 @@
+"""utils.py"""
+# -*- coding: utf-8 -*-
+from datetime import datetime, timedelta
+from django.db.models import OuterRef, Subquery, F
+
+from papersfeed.models.papers.paper import Paper
+from papersfeed.models.users.user import User
+from papersfeed.models.users.user_action import UserAction
+from papersfeed.models.users.user_recommendation import UserRecommendation
+
+def select_user_actions(_):
+    """Select user actions"""
+    new_actions = UserAction.objects.filter(
+        modification_date__gt=(datetime.now() + timedelta(days=-1))
+    ).annotate(
+        UserId=__get_user_id('user'),
+        ItemId=__get_paper_id('paper'),
+        Type=F('type'),
+        Count=F('count'),
+    ).values(
+        'UserId', 'ItemId', 'Type', 'Count'
+    )
+
+    new_actions = list(new_actions)
+
+    for action in UserAction.objects.all():
```

Comment:
Just try `new_actions.update(count=0)` and don't covert `new_actions` to `list`.
You should refer to Django docs like [here](https://docs.djangoproject.com/en/2.2/ref/models/querysets/). In many cases, Django supports smart and efficient way than naive Python codes.

Reply:
new_actions is a queryset with dictionaries, not objects(because of values), so errors occur, when I use update for it. Also, I should change queryset to list, because of json serialization.

Reply:
But I think I should reduce the number of update actions!

Reply:
Got it.

Reply:
How about now?

Reply:
What's the changes? It's hard to compare on GitHub.

Reply:
Is that the code I pointed in https://github.com/swsnu/swpp2019-team3/pull/122#discussion_r350731256? Could you explain the meaning?